### ISSUE TITLE
Chaos test OM

### ISSUE THREAD

**Issue by Laremere** [googleforgames/open-match#1106]
Chaos testing is the practice of deliberately inserting errors into the system to ensure that the system is able to recover from errors.

For example, https://github.com/berkay-dincer/kubethanos will delete a bunch of pods in a cluster.
It would be good to chaos test Open Match so we know that it is good at recovering from node or other types of failures.

**Comment by Laremere** [comment]
I did a preliminary chaos test to see how Open Match handles recovery from a downed match function.  In this scenario, the match function deployment is scaled to 0 for one minute, and then scaled back up to 3 for 7 minutes.  Grafana is observed to look for any errors.

Create Cluster
```
gcloud container clusters create "scale-script" --zone "us-west1-a" --no-enable-basic-auth --machine-type "n1-standard-32" --num-nodes "3"
```

Auth cluster
```
gcloud container clusters get-credentials scale-script --zone us-west1-a
```

Push images
```
make push-images -j20
```

Install helm chart
```
make install-scale-chart
```

Watch Grafana scale dashboard
```
 kubectl port-forward --namespace open-match service/open-match-grafana 3000:3000
```

Scale match function up and down
```
while [ 1 ]; do   kubectl scale deployment --namespace open-match om-function --replicas 0;   sleep 1m;   kubectl scale deployment --namespace open-match om-function --replicas 3;   sleep 7m; done
```

Results:
Things work as expected ¯\_(ツ)_/¯

Match match errors occur during the minute when the match function is down (as expected), and once it’s back up the extra tickets created all find matches.

Zoomed in 15 minute few showing a few cycles in detail.
![image](https://user-images.githubusercontent.com/202599/89374009-0a597380-d69f-11ea-9581-359301c9a0ca.png)

6 hour view showing that I ran this for several hours without problem.  (a few at the beginning were done manually, so are less regular)
![image](https://user-images.githubusercontent.com/202599/89374020-0fb6be00-d69f-11ea-8d3a-93a8733af60c.png)
